contracts/GnosisSafe.sol

```diff
@@ -151,7 +151,7 @@ contract GnosisSafe is MasterCopy, BaseSafe, SignatureDecoder, SecuredTokenTrans
             // If v is 0 then it is a contract signature
             if (v == 0) {
                 // When handling contract signatures the address of the contract is encoded into r
-                currentOwner = address(r);
+                currentOwner = address(uint256(r));
```

Comment:
it is important here to cast to `uint256` and then to `address`. Even so another possible conversion is first to `bytes20` and then to `address`, this would result in cut of addresses since `bytesX` use different padding